embassy-usb-driver/src/lib.rs

```diff
@@ -349,6 +365,20 @@ pub trait ControlPipe {
 pub trait EndpointIn: Endpoint {
     /// Write a single packet of data to the endpoint.
     async fn write(&mut self, buf: &[u8]) -> Result<(), EndpointError>;
+
+    /// Write all the data from buf to the endpoint one wMaxPacketSize chunk at a time.
+    ///
+    /// If the buffer size is evenly divisible by wMaxPacketSize, this will also ensure the
+    /// terminating zero-length-packet is transmitted.
+    async fn write_data(&mut self, buf: &[u8]) -> Result<(), EndpointError> {
+        for chunk in buf.chunks(self.info().max_packet_size as usize) {
+            self.write(chunk).await?;
+        }
+        if buf.len() % self.info().max_packet_size as usize == 0 {
+            self.write(&[]).await?;
+        }
```

Comment:
Whether you send the ZLP or not depends on whether the host is expecting one or not, so there should be a param for it.

Reply:
done